src/pages/index.js

```diff
@@ -5,52 +5,281 @@ import PopupWithForm from "../components/PopupWithForm.js";
+);
+deleteConfirmModal.setEventListeners();
+
+let cardToDelete = null;
+
+// ----------------------------
+
+// Add event listeners to open the modal when delete button is clicked
+
+function handleDeleteClick(card) {
+  deleteConfirmModal.open(card);
+  // cardToDelete = card;
+}
+
+const deleteSubmitButton = deleteConfirmForm.querySelector(
+  "button[type='submit']"
+);
+
+function handleCardDeleteSubmit(card) {
+  deleteSubmitButton.textContent = "Deleting...";
+  const card_id = card._id;
+  console.log("Card ID:", card_id); // Debugging log
+
+  api
+    .deleteCard(card_id)
+    .then((message) => {
+      console.log(message);
+      deleteConfirmModal.close();
+      card.handleDeleteClick(); // Assuming this method exists to remove the card from the DOM
+    })
+    .catch((error) => {
+      console.error("Error deleting card:", error);
+    })
+    .finally(() => {
+      deleteSubmitButton.textContent = "Yes"; // Reset button text after completion
+    });
+}
+
+// ----------------------------
+
+// like button
+// Function to handle like click
 
-const handleProfileFormSubmit = (formData) => {
-  userInfo.setUserInfo({ name: formData.title, job: formData.description });
-  profileForm.reset();
-  editProfilePopup.close();
-};
+function findCardById(cardId) {
+  return initialCards.find((card) => card._id === cardId || card.id === cardId);
+}
+function handleLikeClick(cardData) {
+  if (cardData._isLiked) {
+    api
+      .removeLike(cardData._id)
+      .then((updatedCard) => {
+        cardData.setLike(false); // Update local like status
+      })
+      .catch((err) => console.log(err));
+  } else {
+    api
+      .addLike(cardData._id)
+      .then((updatedCard) => {
+        cardData.setLike(true); // Update local like status
+      })
+      .catch((err) => console.log(err));
+  }
+  // cardData.renderLikes();
+}
+
+// Function to update card likes in the UI
+function updateCardLikes(cardId, likes) {
+  const cardElement = document.querySelector(`.card[data-id="${cardId}"]`); // Make sure card elements have data-id attribute
+  console.log("cardElement:", cardElement); // Log cardElement
+
+  // Check if cardElement is found
+  if (cardElement === null) {
+    console.error(`Card element not found for id: ${cardId}`);
+    return;
+  }
+  const likeCountElement = cardElement.querySelector(".card__like-count");
+
+  likeCountElement.textContent = likes.length;
+
+  const likeButton = cardElement.querySelector(".card__button-like");
+  if (likes.some((user) => user._id === currentUser._id)) {
+    likeButton.classList.add("card__button-like_active");
+  } else {
+    likeButton.classList.remove("card__button-like_active");
+  }
+}
+
+// ----------------------------
+
+const profileSubmitButton = document.querySelector("#profile-submit-button");
+
+function handleProfileFormSubmit(event) {
+  // event.preventDefault();
+  profileSubmitButton.textContent = "Saving...";
+
+  const formData = new FormData(profileForm);
```

Comment:
`  const formData = new FormData(profileForm);
`

Only method `_getInputValues` collects the whole data of the inputs **inside** `PopupWithForm` and passes it to the call of the `submit handler`. So you need to use the collected data rather than the inputs.

Reply:
i am sorry but i am having hard time understanding this

Reply:
> i am sorry but i am having hard time understanding this

You have method `_getInputValues` inside `PopupWithForm` which collects the whole data from the inputs. You get this data in the call


```js
function handleProfileFormSubmit(data) {  // data comes here
  profileSubmitButton.textContent = "Saving...";

  const formValues = {
    name: data.title,  // use it here
    about: data.description, // use it here
  };


```